src/UI/NavigationContainer.elm

```diff
@@ -168,31 +318,92 @@ stateUpdate msg state =
 -- Constructors
 
 
+{-| The correct way of instantiating a [`Nav.State`](UI-NavigationContainer#State).
```

Comment:
I think `Nav.State` must be opaque too, this way, if we add more features, it is less probably to be a breaking change.